### ISSUE TITLE
Search relevance doesn't seem great

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vsmarketplace#71]
I build the only raster image editor for VS Code, when I search for "image editor" almost all the results of the first page are not relevant:

![image](https://user-images.githubusercontent.com/2193314/116878070-a59e5b80-abd3-11eb-802c-1f32c32e3dd8.png)

My extension is the 183rd entry:

![image](https://user-images.githubusercontent.com/2193314/116878034-9a4b3000-abd3-11eb-924f-26ae10ebb539.png)

Am I doing something wrong? It seems like the search is prioritizing "editor" and a high download count over actually matching the word "image" or the phrase "image editor". Ensuring both words show up in the search result and that the phrase appears together seems like the most important thing for relevance.

**Comment by RandomFractals** [comment]
yeah, you are not missing anything. vs marketplace search is really bad. works only with direct hits on extension name, doesn't check extension description or keywords much and load lots of old extensions with more installs first, rather than relevant extensions.

It's been frustrating to use it and we've seen a number for complains from devs just searching for the right extension and ending up asking as in vscode-dev-community slack.

**Comment by eamodio** [comment]
This is a LONG standing issue, and here are some related issues:

https://github.com/microsoft/vsmarketplace/issues/19
https://github.com/microsoft/vscode/issues/42890
https://github.com/microsoft/vscode/issues/73182
https://github.com/microsoft/vscode/issues/106418
https://github.com/microsoft/vscode/issues/102193
https://github.com/microsoft/vscode/issues/29779
https://github.com/microsoft/vscode/issues/51207
https://github.com/microsoft/vscode/issues/69810

**Comment by anangaur** [comment]
Can this be duped to #154 after ensuring its listed as a category of search improvements we want to pursue? /cc: @prashantvc @isidorn

**Comment by Tyriar** [comment]
@anangaur #154 links to this issue in its top comment, it looks like you're using #154 as an overall checklist in which case this should stay open until it's improved?
